esp/templates/program/modules/studentonsite/schedule.html

```diff
@@ -0,0 +1,80 @@
+</head>
+
+<meta name="viewport" content="width=device-width, initial-scale=1.0">
+
+<body>
+
+<div class="navbar">
+  <table>
+    <tr>
+      <td class="selected"><a class="no-dec" href="/learn/{{one}}/{{two}}/studentonsite/" style="display:block;"><i class="material-icons md-5">event_note</i>
+        <br>Schedule</td>
+      <td><a class="no-dec" href="/learn/{{one}}/{{two}}/onsitemap" style="display:block;"><i class="material-icons md-5">map</i>
+        <br>Map</a></td>
+      <td><a class="no-dec" href="/learn/{{one}}/{{two}}/onsitecatalog" style="display:block;"><i class="material-icons md-5">event</i>
+        <br>Catalog</td>
+      <td><a class="no-dec" href="/learn/{{one}}/{{two}}/onsitesurvey" style="display:block;"><i class="material-icons md-5">question_answer</i>
+        <br>Survey</td>
+    </tr>
+  </table>
+</div>
+
+<center>
+
+  <div class="main">
+    <table class="studentschedule">
+      <tbody>
+        <th colspan="3" class="user_info">
+            <i>Classes for {{user.first_name}} {{user.last_name}} - ID: {{user.id}}</i>
+        </th>
+        {% for timeslot in timeslots %}
+            {% ifchanged timeslot.0.start.day %}
+                <tr>
+                    <th colspan="3" class="day">{{ timeslot.0.pretty_date }}</th>
+                </tr>
+            {% endifchanged %}
+            <tr class="schedule_row">
+            {% with timeslot.1.0 as cls %}
+                <td class="time">{{ timeslot.0.short_time }}</td>
+                <td class="cls">
+                {% if cls %}
+                    <!--link to course details?-->
+                    <b>{{ cls.section.title }}{% if not cls.first_meeting_time %} (continued){% endif %}</b>
+                {% else %}
+                    No class
+                {% endif %}
+                {% if checked_in %}
+                    <div class="location">
+                        <i>{{ cls.section.prettyrooms|join:", " }}</i>
+                    </div>
+                {% endif %}
+                </td>
+                <td class="edit">
+                    {% if checked_in %}
+                        <!--link to filtered course catalog like the fillslot page-->
+                        <a class="no-dec" href="/learn/{{one}}/{{two}}/onsitecatalog/{{timeslot.0.id}}"><i class="material-icons md-4">{% if cls %}create{% else %}add{% endif %}</i></a>
+                        {% if cls %}
+                            <a class="no-dec" href="/learn/{{one}}/{{two}}/onsiteclearslot/{{timeslot.0.id}}"><i class="material-icons md-4">clear</i></a>
+                        {% endif %}
+                    {% else %}
+                        <!--otherwise, grey out icon-->
+                        <i class="material-icons md-4 greyed">{% if cls %}create{% else %}add{% endif %}</i>
+                        {% if cls %}
+                            <i class="material-icons md-4 greyed">clear</i>
```

Comment:
I think the greyed out icon was pretty non-obvious in terms of its usage; I tried clicking it for a while before I realized/remembered I had to be checked in. :P Maybe just no icon here? I don't know what the greyed out icon is supposed to communicate to the student...

Reply:
Hmm, wouldn't no icon look weird, because there'd be a big gap (as it is, it's all nice and symmetrical)? In chatting, you suggested adding a popup, which I think could be a nice alternative!